src/scene_map.cpp

```diff
@@ -200,6 +200,7 @@ void Scene_Map::TransitionOut(SceneType next_scene) {
 			// If screen is already erased, RPG_RT does nothing for 40 frames.
 			transition.InitErase(Transition::TransitionNone, this, 40);
 		}
+		Main_Data::game_screen->CancelBattleAnimation();
```

Comment:
This breaks an edge case where when you start a battle before the animation completes, the flash and sound effects of the animation will occur during the battle.

Reply:
The cause is that the CancelBattleAnimation destroys the Animation but Drawable->Take fails to take it because it is not on the Battle stack. After returning to the map the Drawable list contains a dangling reference to this animation and crashes.

Will sharing the BattleAnimation help? Only must be set to Hidden (at the same location where I put the picture update code)

Reply:
Imo the "Drawable->Take" when Deregistering should assert when the Take failed to catch such bugs but this breaks most Scene destructions.

Reply:
Sharing and hiding should work. That will still make sound effects and screen effects occur. Ideally we would be better off to just not allocate and destroy the animation every time. Have it's lifetime match the lifetime of screen. That would also be a minor performance win.

> Imo the "Drawable->Take" when Deregistering should assert when the Take failed to catch such bugs but this breaks most Scene destructions.

The whole lifetime model of drawables is flawed. The new code makes that even more explicit. We need another refactor which better ties drawable lifetimes to scenes / lists.

Reply:
I solved this now via delayed deletion because there is another way to crash that is not solvable through sharing: When you go from Debug scene to Battle while an animation is active (sharing doesn't work because Scene Map is not the stack top). Is this ok?

Reply:
Ouch, that means there is another bug. Battle pictures and weather won't work when you trigger battles from the debug menu. 

We can punt that as it isn't critical https://github.com/EasyRPG/Player/issues/2120

This workaround is for by me for now.